### DISCUSSION TITLE
How to remove views?

### DISCUSSION THREAD

**Discussion by bengraves-five** [eclipse-theia/theia#8293] [General]
I want to remove the following views from my IDE:

- Call Hierarchy
- Debug
- Debug Console
- Extensions
- History
- Output
- Plugins
- Test
- Type Hierarchy

How can I remove these views or at least hide them in the View menu?

Thanks!

**Comment by vince-fugnitto** [comment]
@bengraves-five since you are removing most views from the framework you should probably think about if the extensions which contribute these views are needed for your use-case, and if so remove the views themselves through dependency injection or remove the menu item which toggles/opens the views. 

What exactly is your use-case?

**Comment by benoitf** [comment]
hi,

if you build your own assembly of theia, you could remove extensions that provide these views

else you may unbind stuff from a custom extension or hide.... but this looks more strange to hide stuff instead of removing it

**Comment by tortmayr** [comment]
Hi, 
I 'm having a similar issue. I have already removed all unwanted extensions from my application (only using @theia/core & @theia/preferences). However via transitive dependencies I still pull in a lot of other extensions (monaco,outlineview,languages etc.). 

What would be the best way to hide some of these unwanted UI contributions?
@benoitf mentioned the possibility of unbinding stuff from a custom extension but this doesn't seem to work for multi-injections.

Thanks!

**Reply by akosyakov** [reply]
```
// remove `Call Hierarch` view contribution
    rebind(CallHierarchyContribution).toConstantValue({
        registerCommands: () => { },
        registerMenus: () => { },
        registerKeybindings: () => { }
    } as any);
```

Each extension has usually a single contribution implementing various application contribution interfaces. You can stub such contributions that they do nothing.

It is obviously not optimal, and getting rid of all code at runtime would be the best. We unfortunately end up in this situation by chasing VS Code extensions support. There is an issue: https://github.com/eclipse-theia/theia/issues/7754 Please up vote there.
